### PR TITLE
fix(#176): set line to '0' in `empty-object` if `@line` is weird

### PR DESCRIPTION
**Pull request by h1alexbel**
In this pull request I fixed line number reporting in `empty-object` lint, in cases when line number is NaN.

ref #176
History:
- **fix(#176): tests**
- **fix(#176): line 0 if NaN**

### CODE REVIEW

**Comment by h1alexbel** [comment]
@maxonfjvipon please check

File: src/main/resources/org/eolang/lints/errors/empty-object.xsl [hunk 1]
```diff
@@ -29,7 +29,7 @@ SOFTWARE.
       <xsl:for-each select="//o[not(o)]">
         <xsl:element name="defect">
           <xsl:attribute name="line">
-            <xsl:value-of select="if (@line) then @line else '0'"/>
+            <xsl:value-of select="if (@line and not(number(@line) = NaN) and number(@line) = number(@line)) then @line else '0'"/>
```
Comment (maxonfjvipon) [inline src/main/resources/org/eolang/lints/errors/empty-object.xsl:32]: @h1alexbel as far as I know `NaN = NaN `is `false`, so if you checked that line is not `NaN`, what this `line = line` extra comparison for?

**Comment by h1alexbel** [comment]
@maxonfjvipon updated. Take a look, please

**Review by maxonfjvipon — APPROVED** [review]

**Comment by h1alexbel** [comment]
@yegor256 take a look, please

File: src/main/resources/org/eolang/lints/errors/empty-object.xsl [hunk 2]
```diff
@@ -29,7 +29,7 @@ SOFTWARE.
       <xsl:for-each select="//o[not(o)]">
         <xsl:element name="defect">
           <xsl:attribute name="line">
-            <xsl:value-of select="if (@line) then @line else '0'"/>
+            <xsl:value-of select="if (@line and number(@line) = number(@line)) then @line else '0'"/>
```
Comment (yegor256) [inline src/main/resources/org/eolang/lints/errors/empty-object.xsl:32]: @h1alexbel why do we check it only in this XSL? If this problem is external, we should be careful with the `@line` attribute in all XSL stylesheets. Maybe let's introduce a function `eo:lineno(x)`, which will do what you do here, but used in all XSLs?

**Review by yegor256 — CHANGES_REQUESTED** [review]
@h1alexbel thanks! see one comment above

**Comment by h1alexbel** [comment]
@yegor256 updated. Take a look, please

**Comment by yegor256** [comment]
@h1alexbel since you introduced this new function, maybe it's reasonable to use it in all XSLs? Also, no need to start the name of this XSL with underscore. It was done in EO repository in order to exclude this XSL from test pack.

**Comment by h1alexbel** [comment]
@yegor256 updated. Take a look, please

**Review by yegor256 — APPROVED** [review]
@h1alexbel perfect!

**Pull request merged**